public_dropin_gpu_environments/nemollm_inference_ms/Dockerfile

```diff
@@ -0,0 +1,84 @@
+# This is the default base image for use with user models and workflows.
+FROM nvcr.io/ohlfw0olaadg/ea-participants/nemollm-inference-ms:24.01
+USER root
+# Need to change the UID of nemo user until upstream fixes this as
+# it is out-of-range for our rootless buildkit image builder.
+RUN groupmod -g 1000 nemo && \
+  usermod -u 1000 -g 1000 nemo && \
+  chown -R 1000.1000 /home/nemo /model-store && \
+  rm /var/log/lastlog
+
+# Need to do this to squash our hack above so rootless buildkit won't
+# try and unpack any layers with out-of-bound UIDs
+FROM scratch
+COPY --from=0 / /
```

Comment:
FWIW, this was an assumption that I never tested. Now that we are out of the time crunch for the demo, it is probably worth validating this is necessary because it would simplify things if it wasn't needed.

Reply:
without layers squash the image builder fails:
```
{"timestamp":"2024-03-20 18:04:58.294","level":"INFO","thread":"pool-7-thread-1","logger":"com.datarobot.imagebuilder.util.Logger","message":"#7 ERROR: mount callback failed on /run/user/1000/containerd-mount2453652930: failed to Lchown \"/run/user/1000/containerd-mount2453652930/home/nemo\" for UID 4474987, GID 4474987: lchown /run/user/1000/containerd-mount2453652930/home/nemo: invalid argument (Hint: try increasing the number of subordinate IDs in /etc/subuid and /etc/subgid)","context":"default"}
```

Reply:
https://kibana-o11y.aws.us.stg.int.datarobot.com/app/r/s/05Ldj?auth_provider_hint=anonymous1